src/index.js

```diff
@@ -70,8 +78,85 @@ class HMIApp extends React.Component {
     handleDDToggle(){
         store.dispatch(setDDState(!this.props.dd));
     }
+    pickResolution(event) {
+        var match = event.target.value.match(/(\d+)x(\d+) Scale (\d+.?\d*)/);
+        var capability = {
+            systemCapabilityType: 'VIDEO_STREAMING',
+            videoStreamingCapability: {
+                scale: parseFloat(match[3]),
+                preferredResolution: {
+                    resolutionWidth: parseInt(match[1]),
+                    resolutionHeight: parseInt(match[2])
+                }
+            }
+        }
+        this.setState({ resolution: event.target.value, scale: match[3] });
+
+        bcController.onSystemCapabilityUpdated(capability, this.props.activeAppId);
+    }
+
+    onTouchEvent(type, event) {
+        if (!this.videoRect) {
+            var video = document.getElementById('navi_stream');
+            this.videoRect = video.getBoundingClientRect();
+        }
+
+        if (event.pageX < this.videoRect.left
+            || event.pageX > this.videoRect.right
+            || event.pageY < this.videoRect.top
+            || event.pageY > this.videoRect.bottom) {
+            return;
+        }
+
+        uiController.onTouchEvent(type, [{
+            id: 0,
+            ts: [ parseInt(event.timeStamp) ],
+            c: [{
+                x: event.pageX - this.videoRect.x,
+                y: event.pageY - this.videoRect.y
+            }]
+        }]);
+    }
+
+    onTouchBegin(event) {
+        this.touchInProgress = true;
+        this.onTouchEvent('BEGIN', event);
+    }
+
+    onTouchMove(event) {
+        if (!this.touchInProgress) {
+            return;
+        }
+
+        this.onTouchEvent('MOVE', event);
+    }
+
+    onTouchEnd(event) {
+        if (!this.touchInProgress) {
+            return;
+        }
+
+        this.touchInProgress = false;
+        this.onTouchEvent('END', event);
+    }
+
     render() {
         const themeClass = this.state.dark ? 'dark-theme' : 'light-theme';
+        var videoStyle = { position: 'absolute', width: 960, height: 600, top: 75, left: 0, backgroundColor: 'transparent', objectFit: 'fill' };
```

Comment:
Might want to try importing the master height / width and setting video style that way.

Reply:
done in https://github.com/smartdevicelink/generic_hmi/pull/345/commits/6125f5abb20538a97e15a641fabaa88dc9a1c3de and also subtracted the appheader height in https://github.com/smartdevicelink/generic_hmi/pull/345/commits/97519be979874f42cb47739822067a6a536c4bea